modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Sorting.tsx

```diff
@@ -0,0 +1,387 @@
+					),
+					type: 'danger',
+				});
+
+				return null;
+			}
+
+			const fieldsArray: Array<string> = [];
+
+			const isObjectSchema =
+				responseJSON.components.schemas[restSchema].xml.name ===
+				'ObjectEntry';
+
+			Object.keys(properties).forEach((propertyKey) => {
+				const propertyValue = properties[propertyKey];
+
+				if (isObjectSchema && !propertyValue.extensions) {
+					return;
+				}
+
+				if (propertyKey === 'x-class-name') {
+					return;
+				}
+
+				const type = propertyValue.type;
+
+				if (type === 'object' || type === 'array') {
+					return;
+				}
+
+				fieldsArray.push(propertyKey);
+			});
+
+			setFields(fieldsArray);
+		};
+
+		getFields();
+	}, [fdsView]);
+
+	return (
+		<div className="fds-view-fields-modal">
+			<ClayModal.Header>
+				{Liferay.Language.get('new-default-sort')}
+			</ClayModal.Header>
+
+			<ClayModal.Body>
+				<ClayForm.Group>
+					<label htmlFor="field">
+						{Liferay.Language.get('field')}
+
+						<RequiredMark />
+					</label>
+
+					<DropDown
+						closeOnClick
+						menuWidth="sm"
+						trigger={
+							<ClayButton
+								className="form-control form-control-select"
+								displayType="unstyled"
+							>
+								{field ||
+									Liferay.Language.get('choose-an-option')}
+							</ClayButton>
+						}
+					>
+						<DropDown.ItemList>
+							{fields?.map((field) => (
+								<DropDown.Item
+									key={field}
+									onClick={() => setField(field)}
+								>
+									{field}
+								</DropDown.Item>
+							))}
+						</DropDown.ItemList>
+					</DropDown>
+				</ClayForm.Group>
+
+				<ClayForm.Group>
+					<label htmlFor="sorting">
+						{Liferay.Language.get('sorting')}
+
+						<RequiredMark />
+					</label>
+
+					<DropDown
+						closeOnClick
+						menuWidth="sm"
+						trigger={
+							<ClayButton
+								className="form-control form-control-select"
+								displayType="unstyled"
+							>
+								{sorting.label}
+							</ClayButton>
+						}
+					>
+						<DropDown.ItemList>
+							{SORTING_OPTIONS.map((sortingOption) => (
+								<DropDown.Item
+									key={sortingOption.value}
+									onClick={() => setSorting(sortingOption)}
+								>
+									{sortingOption.label}
+								</DropDown.Item>
+							))}
+						</DropDown.ItemList>
+					</DropDown>
+				</ClayForm.Group>
+			</ClayModal.Body>
+
+			<ClayModal.Footer
+				last={
+					<ClayButton.Group spaced>
+						<ClayButton
+							displayType="secondary"
+							onClick={() => closeModal()}
+						>
+							{Liferay.Language.get('cancel')}
+						</ClayButton>
+
+						<ClayButton
+							disabled={!field}
+							onClick={() => {
+								closeModal();
+
+								onSave();
+							}}
+						>
+							{Liferay.Language.get('save')}
+						</ClayButton>
+					</ClayButton.Group>
+				}
+			/>
+		</div>
+	);
+};
+
+const Sorting = ({fdsView, fdsViewsURL}: FDSViewSectionInterface) => {
+	const [fdsSorting, setFDSSorting] = useState<Array<FDSSortingType> | null>(
```

Comment:
If we say array _or_ null, that means we will also have to always have a check before we call any array methods. Can we just guarantee this is an array by defaulting the value to []? Is there any necessity of having a null value?

Reply:
I see, thanks for pointing that out. I'll switch to defaulting to `[]`.